lib/app/features/posts/data/favorites_repository.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:developer';
+
+import 'package:black_tax_and_white_benefits/app/features/posts/data/database.dart';
+import 'package:black_tax_and_white_benefits/app/features/posts/domain/post.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'favorites_repository.g.dart';
+
+@riverpod
+class FavoritesRepository extends _$FavoritesRepository {
```

Comment:
thius shouldn't be a notifier because there's really no state, you just need
@riverpod
Repository(RepositoryRef ref) => RepositoryRef();  or soething like that
adn int he repository
final Ref;
and the constructor
Repository(this.ref)

Reply:
Oh that makes sense, I didn't realize I was generating Change Notifiers. Is there a way to do this using annotation? Wish I could keep it consistent with annotation vs. manually defining providers